src/backend.rs

```diff
@@ -489,6 +489,13 @@ where
     }
 }
 
+/// mode for the `SharedBackend::do_get_storage` function
+#[derive(Clone, Debug, PartialEq)]
+pub enum Mode {
+    Cache,
```

Comment:
hmm, what I had in mind for this pr is that this setting determines whether we're using

```
tokio::task::block_in_place
```

or not. I think we can do 

```rust
pub enum BlockingMod {
  BlockInPlace
  Block
}
```